### DISCUSSION TITLE
ipxe and Cisco C240 M5L with VIC1455 not working

### DISCUSSION THREAD

**Discussion by mdn19866** [ipxe/ipxe#537] [General]
Hello everyone,

We have problems getting iPXE to boot with UEFI boot mode working on a Cisco C240M5L standalone server.
If we choose UEFI boot mode and PXE boot, the PXE starts, gets a station IP, but then just does nothing and jumps into the next line of the boot order. In the capture on the upstream switch, we see two consequent downloads of undionly.kpxe file over tftp and the boot process is aborted with no clear error message.

With legacy BIOS boot chosen, the iPXE boot just works fine. The issue is only with UEFI boot mode.

We saw already an issue with another Cisco server here in the list -> https://github.com/ipxe/ipxe/issues/385

Thanks for your help in advance.

**Comment by NiKiZe** [comment]
Does not seem to be an issue/bug with iPXE itself, but rather an user issue, so moved to discussions.

You mention undionly.kpxe - that is a binary for (legacy) pcbios, while you mention EFI.
that binary will never run in EFI mode, and what you see is then expected.

Change your dhcp config to instead send ipxe.efi or snponly.efi when there is an request from a EFI client.

You did not include any information on how your DHCP is setup, please post details about that if you want further assistance.

**Reply by mdn19866** [reply]
Thanks for the push in the right direction. We had seen it, but had not taken it for serious. It was indeed a misconfiguration on the DHCP.
